stubs/openpyxl/openpyxl/worksheet/dimensions.pyi

```diff
@@ -1,95 +1,118 @@
-from _typeshed import Incomplete
+from _typeshed import Incomplete, Unused
+from collections.abc import Callable, Generator
+from typing import Generic, TypeVar
+from typing_extensions import Self
 
 from openpyxl.descriptors import Strict
+from openpyxl.descriptors.base import Alias, Bool, Float, Integer, String
 from openpyxl.descriptors.serialisable import Serialisable
 from openpyxl.styles.styleable import StyleableObject
 from openpyxl.utils.bound_dictionary import BoundDictionary
+from openpyxl.worksheet.worksheet import Worksheet
+from openpyxl.xml.functions import Element
+
+_DimT = TypeVar("_DimT", bound=Dimension)
 
 class Dimension(Strict, StyleableObject):
-    __fields__: Incomplete
-    index: Incomplete
-    hidden: Incomplete
-    outlineLevel: Incomplete
-    outline_level: Incomplete
-    collapsed: Incomplete
-    style: Incomplete
+    __fields__: tuple[str, ...]
+
+    index: Integer
+    hidden: Bool
+    outlineLevel: Integer
+    outline_level: Alias
+    collapsed: Bool
+    style: Alias
+
     def __init__(
-        self, index, hidden, outlineLevel, collapsed, worksheet, visible: bool = ..., style: Incomplete | None = ...
+        self,
+        index: int,
+        hidden: bool,
+        outlineLevel: int | None,
+        collapsed: bool,
+        worksheet: Worksheet,
+        visible: bool = True,
+        style: Incomplete | None = None,
     ) -> None: ...
-    def __iter__(self): ...
-    def __copy__(self): ...
+    def __iter__(self) -> Generator[tuple[str, str], None, None]: ...
```

Comment:
I'd prefer the simpler annotation here, as it seems to me that it's an implementation detail that this returns a generator as opposed to some other kind of iterator. But I won't insist :)

```suggestion
    def __iter__(self) -> Iterator[tuple[str, str]]: ...
```